lib/backend/k8s/k8s.go

```diff
@@ -79,12 +98,91 @@ func NewKubeClient(kc *KubeConfig) (*KubeClient, error) {
 	}
 
 	// Create the clientset
-	cs, err := clientset.NewForConfig(config)
+	cs, err := kubernetes.NewForConfig(config)
 	if err != nil {
 		return nil, err
 	}
 	log.Debugf("Created k8s clientSet: %+v", cs)
-	return &KubeClient{clientSet: cs}, nil
+
+	tprClient, err := buildTPRClient(config)
+	if err != nil {
+		return nil, err
+	}
+	kubeClient := &KubeClient{
+		clientSet: cs,
+		tprClient: tprClient,
+	}
+
+	// Ensure the necessary ThirdPartyResources exist in the API.
+	err = kubeClient.ensureThirdPartyResources()
+	if err != nil {
+		return nil, goerrors.New(fmt.Sprintf("Failed to create necessary ThirdPartyResources: %s", err))
+	}
+
+	return kubeClient, nil
+}
+
+// TODO: The way this is right now, we'll run this every time
+// a client is created.  This is OK for felix, but less nice for say,
+// calicoctl.
+func (c *KubeClient) ensureThirdPartyResources() error {
+	// Ensure a resource exists for Calico global configuration.
+	tpr := extensions.ThirdPartyResource{
+		ObjectMeta: kapiv1.ObjectMeta{
+			Name:      "global-config.projectcalico.org",
+			Namespace: "kube-system",
+		},
+		Description: "Calico Global Configuration",
+		Versions:    []extensions.APIVersion{{Name: "v1"}},
+	}
+	_, err := c.clientSet.Extensions().ThirdPartyResources().Get(tpr.ObjectMeta.Name)
+	if err != nil {
+		// The resource needs to be created.
+		_, err := c.clientSet.Extensions().ThirdPartyResources().Create(&tpr)
```

Comment:
Rather than getting and then creating, it's better to just create and handle the appropriate error that indicates it's already created.  It's one less round trip on the first create, but more importantly closes the window where two concurrent threads could attempt to create the resource and expect to pass.